16ALU/16ALU.luc/source/alu16.luc

```diff
@@ -9,33 +9,47 @@ module alu16 (
   ) {
 
   // Initialise all the submodules //
+  //to REPLACE submod names with actual names !!!
   // ADD/SUB
+  adder addmod;
   
   // BOOL
+  boolean boole;
   
   // SHIFT
   shifter shifter;
   
   // CMP
+  compare comp;
   
-  ///////////////////////////////
-  
+ 
   always {
     
     // Connect the input for all the submodules //
     // ADD/SUB
     
+    addmod.a = a;
+    addmod.b = b;
+    addmod.alufn = alufn[0];
+    
     // BOOL
+    boole.a = a;
+    boole.b = b;
+    boole.alufn = alufn[3:0]
     
     // SHIFT
     shifter.a = a;
     shifter.b = b[3:0];
     shifter.alufn = alufn[1:0];
     
     // CMP
-    //////////////////////////////////////////////
+    comp.alufn = alufn[2:1];
+    
+    comp.z = addmod.z
+    comp.v = addmod.v
+    comp.n = addmod.n
     
-    // Define the default output for z, v, n
+    // Define the default output for z, v, n //should this part go below? to the output defult?
```

Comment:
Remove the comment.
z,v,n need to declare outside of the case as only the add module is using the z,v,n and we will only go to default case if the alufn does not match